lib/masamune/commands/hive.rb

```diff
@@ -165,8 +165,14 @@ def command_args_for_simple_file
 
     def command_args_for_template
       @rendered_file = Masamune::Template.render_to_file(@file, @variables)
-      filesystem.copy_file_to_dir(@rendered_file, filesystem.get_path(:tmp_dir))
-      ['-f', filesystem.get_path(:tmp_dir, File.basename(@rendered_file))]
+      filesystem.copy_file_to_file(@rendered_file, remote_file)
+      ['-f', remote_file]
+    end
+
+    private
+
+    def remote_file
+      @remote_file ||= File.join(filesystem.mktempdir!(:tmp_dir), filesystem.basename(@file)).gsub(/.erb\z/,'')
```

Comment:
upload file to unique tmp directory so files with the same name do not conflict, while trying to keep the file name as human readable